packages/file-asset-apis/src/s3/s3-helpers.ts

```diff
@@ -168,6 +168,12 @@ export async function doesBucketExist(
         if (httpStatusCode === 403) {
             throw new TSError(`User does not have access to bucket "${params.Bucket}"`, { statusCode: 403 });
         }
+        if (
+            httpStatusCode === 400
+            && !validateBucketName(params.Bucket as string)
```

Comment:
I don't know that we should actually validate the bucket name here... you were getting a 400 "UnknownError" which could be anything.  Maybe try consoling the error message... might have to JSON.stringify to see it, check the $metadata maybe,  and figure out where the actual error message is in the error.  Then if it's an invalid bucket name error throw "Invalid bucket name" otherwise  "Error checking if bucket exists".

Reply:
Maybe if it's not an invalid bucket name attach a cause too like... 
```
TSError('Error checking if bucket exists', { 
  context: {
     _cause: err 
}
```

Reply:
Sorry, I played around and see what you mean it gives no info in the error message... you could just keep this simple and throw something generic like below if httpStatusCode === 400, or you could continue with your validation function if you prefer .
```
TSError('Unknown error checking if bucket exists, make sure it's a valid bucket name', { 
  context: {
     _cause: err 
}
```